poc/src/app/common/shared/components/bar-chart-wrapper/bar-chart-wrapper.component.ts

```diff
@@ -61,6 +108,22 @@ export class BarChartWrapperComponent extends ChartWrapperBaseComponent implemen
           [MAIN_DATA_SET]: 'bar',
           [NDC_DATA_SET]: 'spline',
         },
+        onclick: (d, element) => {
+          const createPopup = ({ bbox, barsWidth, eventRectWidth, popupWidth, removeCallback }) => {
```

Comment:
A similar logic and the same functions are used in other components. Refactoring is needed.

Reply:
Fixed